Microsoft.TeamServices.Samples.Client/WorkItemTracking/QueriesSample.cs

```diff
@@ -12,24 +13,401 @@ public class QueriesSample : ClientSample
     {
 
         [ClientSampleMethod]
-        public QueryHierarchyItem GetQueryByName()
+        public List<QueryHierarchyItem> GetListOfQueries()
+        {
+            System.Guid projectId = ClientSampleHelpers.FindAnyProject(this.Context).Id;
+            
+            VssConnection connection = Context.Connection;
+            WorkItemTrackingHttpClient workItemTrackingClient = connection.GetClient<WorkItemTrackingHttpClient>();
+
+            List<QueryHierarchyItem> queries = workItemTrackingClient.GetQueriesAsync(projectId, QueryExpand.None, 1).Result;
+
+            if (queries == null)
+            {
+                Console.WriteLine("No queries found");
+            }
+            else
+            {
+                Console.WriteLine("Queries:");
+
+                foreach(var query in queries)
+                {
+                    Console.WriteLine("  {0} - {1}", query.Name, query.Path);
+                }
+            }
+
+            return queries;
+        }
+
+        [ClientSampleMethod]
+        public QueryHierarchyItem GetQueryOrFolderByFolderPath()
         {
             string project = ClientSampleHelpers.FindAnyProject(this.Context).Name;
-            string queryName = "Shared Queries/Feedback";
+            string queryName = "Shared Queries/Current Iteration";
+
+            VssConnection connection = Context.Connection;
+            WorkItemTrackingHttpClient workItemTrackingClient = connection.GetClient<WorkItemTrackingHttpClient>();
+
+            QueryHierarchyItem query = workItemTrackingClient.GetQueryAsync(project, queryName, null, 2).Result;
+
+            if (query == null)
```

Comment:
Check whether this API throws an exception vs. returning null when the named query can't be found (it should).